packages/popover/src/vaadin-popover.js

```diff
@@ -844,31 +774,16 @@ class Popover extends PopoverPositionMixin(
     }
 
     if (
-      (this.__hasTrigger('focus') && this.__mouseDownInside) ||
       event.relatedTarget === this.target ||
-      this._overlayElement.contains(event.relatedTarget)
+      event.relatedTarget === this._overlayElement ||
+      this.contains(event.relatedTarget)
     ) {
       return;
     }
 
     this.__handleFocusout();
   }
 
-  /** @private */
-  __onOverlayMouseDown() {
-    if (this.__hasTrigger('focus')) {
-      this.__mouseDownInside = true;
```

Comment:
This was added in https://github.com/vaadin/web-components/pull/7656 but I was unable to reproduce the issue after switching to native `popover` - the updated check in `focusout` event `this.contains(event.relatedTarget)` seems to cover this case as well.